lib/authorizer/auth-interface.js

```diff
@@ -0,0 +1,81 @@
+var _ = require('lodash'),
+    createAuthInterface;
+
+/**
+* Creates a wrapper around RequestAuth and provides getters and setters helper functions
+*
+* @param {RequestAuth} auth
+* @constructor
```

Comment:
@harryi3t this is not a constructor. Also, the return type of this should be `AuthInterface`.